### PR TITLE
Update dynamic dropdown docs to include information about dynamic params

### PR DESCRIPTION
**Pull request by charleshtrenholm**
Adds information on how to use new magic variables that provide dynamic dropdown with any given app's current user-defined parameter values at call time.

### CODE REVIEW

File: source/references/UI_spec.rst [hunk 1]
```diff
@@ -217,7 +217,11 @@ similar to the selection of other WS data objects.
 
 :service_params: The parameters supplied to the dynamic service call as JSON. The special text
                  ``"{{dynamic_dropdown_input}}"`` will be replaced by the value from the user input
-                 at call time.
+                 at call time. Additionally, any keywords defined as parameters for the app will be
+                 replaced with the user input value at call time. For instance, if an app has a text
+                 parameter called ``"output_name"`` and and the user has specified the output name
```
Comment (briehl) [inline source/references/UI_spec.rst:222]: Maybe change "called" to "with id" or something similar to make it extra clear we're talking about app parameters by id in spec.json.

File: source/references/UI_spec.rst [hunk 2]
```diff
@@ -217,7 +217,11 @@ similar to the selection of other WS data objects.
...
+                 as "hello_test" in the text field, then the service param ``"{{output_name}}"`` will
+                 be dynamically replaced with the value "hello_test".
```
Comment (briehl) [inline source/references/UI_spec.rst:224]: Also mention that any template that doesn't match a parameter id won't be altered.

**Review by briehl — CHANGES_REQUESTED** [review]
I think this is pretty good. However, I also think this is complicated enough that a complete example might be helpful. It would be nice to see something that goes all the way from some example dynamic_dropdown options, to what the service call would look like, to what gets rendered in the dropdown from the results.

**Review by briehl — APPROVED** [review]
Much better. Thanks!

File: source/references/UI_spec.rst [hunk 3]
```diff
@@ -571,4 +633,7 @@ Each field in the ``parameters`` section can have the following properties:
 
    <a href="https://github.com/kbaseapps/fba_tools/blob/master/ui/narrative/methods/build_metabolic_model/display.yaml" target="_blank">Example</a>
 
+.. |sample_uploader_link| raw:: html
+    <a href="https://github.com/kbaseapps/sample_uploader/blob/master/ui/narrative/methods/filter_samplesets/spec.json#L70" target="_blank">Sample Uploader app</a>
```
Comment (bio-boris) [inline source/references/UI_spec.rst:637]: I think this isn't correctly rendering in the HTML view

Comment (bio-boris) [reply source/references/UI_spec.rst:637]: Its showing vertical bars around the link

**Review by bio-boris — CHANGES_REQUESTED** [review]
Issue with link formatting

File: source/references/UI_spec.rst (hunk 3 above)
Comment (charleshtrenholm) [reply source/references/UI_spec.rst:638]: whoops, missed a line break. Should be fixed now.

Comment (bio-boris) [reply source/references/UI_spec.rst:638]: Can you link to a specific commit in case that file changes?

Comment (bio-boris) [reply source/references/UI_spec.rst:638]: The other links should probably be updated too.

Comment (charleshtrenholm) [reply source/references/UI_spec.rst]: Done, should be good now

**Review by bio-boris — APPROVED** [review]

**Pull request merged**